Enable user to return to application review page

JonF1 opened the pull request:
## Context
Currently the user cannot edit the degree section from the application review page and return there or use backlinks to return there

## Changes proposed in this pull request
* Use the wizard state store to see if user has come from the application review page. If so, return back to application review instead of degree review.
* Add to system specs

## Guidance to review
Manual test to verify that user can edit from application review page and return there or click the backlinks and return to application review page

## Link to Trello card
https://trello.com/c/otSjyZmY/4697-backlinks-return-to-application-review-page

## Things to check

- [ ] If the code removes any existing feature flags, a data migration has also been added to delete the entry from the database
- [ ] This code does not rely on migrations in the same Pull Request
- [ ] If this code includes a migration adding or changing columns, it also backfills existing records for consistency
- [ ] API release notes have been updated if necessary
- [ ] Required environment variables have been updated [added to the Azure KeyVault](/docs/environment-variables.md#deploy-pipeline)

cwrw reviewed (commented):
I was wondering if we could leverage the wizard path history functionality here to avoid having too much logic in the controller: https://github.com/DFE-Digital/apply-for-teacher-training/blob/main/docs/wizards.md#wizardpathhistory where we can also have a dedicated helper for this.  We could also use the wizard store to include the entry point to distinguish where to redirect in the end to avoid using an extra session, let me know your thoughts

JonF1 commented:
> I was wondering if we could leverage the wizard path history functionality here to avoid having too much logic in the controller: https://github.com/DFE-Digital/apply-for-teacher-training/blob/main/docs/wizards.md#wizardpathhistory where we can also have a dedicated helper for this. We could also use the wizard store to include the entry point to distinguish where to redirect in the end to avoid using an extra session, let me know your thoughts

As discussed, we've added a tech debt backlog ticket for a potential rewrite using the path history functionality. And I've updated the PR so that we are using the wizard store instead of the session.

cwrw reviewed (approved):
Only suggestion is to attempt to test different paths of the wizard in unit tests for the wizard/request specs and avoid doing this in scenarios as they are expensive to run
